src/jit/gentree.h

```diff
@@ -5265,6 +5265,13 @@ struct GenTreePutArgStk : public GenTreeUnOp
         return gtNumSlots * TARGET_POINTER_SIZE;
     }
 
+    // Return true if this is a PutArgStk of a SIMD12 struct.
+    // This is needed because such values are re-typed to SIMD16, and the type of PutArgStk is VOID.
+    unsigned isSIMD12()
+    {
+        return (varTypeIsSIMD(gtOp1) && (gtNumSlots == 3));
```

Comment:
@CarolEidt This function looks like it is only correct for architectures with 4 byte slots.  So it looks wrong for ARM64 & AMD64.  Looks like the only caller is _TARGET_X86_, but I think this should be fixed

Reply:
On a 64-bit target, a direct putArgStk of a SIMD12 will actually write 16 bytes. So this only applies for a 32-byte target. So we would need this adjustment for ARM as well.